src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpInteractiveDirectives.cs

```diff
@@ -0,0 +1,284 @@
+                "endif",
+                "endregion",
+                "error",
+                "if",
+                "line",
+                "pragma",
+                "region",
+                "warning");
+
+            VerifyCompletionUnexpectedItemDoesNotExist("cls",
+                "help",
+                "load",
+                "prompt",
+                "reset",
+                "undef",
+                "define");
+        }
+
+        [Fact]
+        public void VerifyHashRDirective()
+        {
+            SubmitText("#r \"System.Numerics\"");
+            SubmitText(@"using System.Numerics;
+var bigInt = new BigInteger();
+bigInt");
+
+            this.WaitForLastReplOutput("[0]");
+        }
+
+        [Fact]
+        public void VerifyLocalDeclarationWithTheSameNameHidesImportedMembersFromHashR()
+        {
+            SubmitText("#r \"System.Numerics\"");
+            SubmitText(@"using System.Numerics;
+class Complex { public int foo() { return 4; } }
+var comp = new Complex();
+comp.foo()");
+
+            this.WaitForLastReplOutput("4");
+        }
+
+        [Fact]
+        public void VerifyLocalDeclarationInCsxFileWithTheSameNameHidesImportedMembersFromHashR()
+        {
+            SubmitText("#r \"System.Numerics\"");
+            SubmitText("using System.Numerics;");
+            using (var temporaryTextFile = new TemporaryTextFile(
+                "directivesScenario4.csx", 
+                "class Complex { public int foo() { return 4; } }"))
+            {
+                temporaryTextFile.Create();
+                SubmitText(string.Format("#load \"{0}\"", temporaryTextFile.FullName));
+                SubmitText(@"var comp = new Complex();
+comp.foo()");
+                this.WaitForLastReplOutput("4");
+            }
+        }
+
+        [Fact]
+        public void VerifyAssembliesReferencedByDefault()
+        {
+            SubmitText(@"using System.Diagnostics;
+Process.GetCurrentProcess().ProcessName");
+            this.WaitForLastReplOutput("\"InteractiveHost\"");
+        }
+
+        [Fact]
+        public void VerifyHashLoadDirective()
+        {
+            using (var temporaryTextFile = new TemporaryTextFile(
+                "directivesScenario6.csx", 
+                "System.Console.WriteLine(2);"))
+            {
+                temporaryTextFile.Create();
+                SubmitText(string.Format("#load \"{0}\"", temporaryTextFile.FullName));
+                this.WaitForLastReplOutput("2");
+                SubmitText("#load text");
+                this.WaitForLastReplOutput("CS7010: Quoted file name expected");
+            }
+        }
+
+        [Fact]
+        public void VerifySquiggleAndErrorMessageUnderIncorrectDirective()
+        {
+            SubmitText("#foo");
+            this.WaitForLastReplOutput("(1,2): error CS1024: Preprocessor directive expected");
+            // TODO implement GetErrorListErrorCount: https://github.com/dotnet/roslyn/issues/18035
```

Comment:
There is a difference between error count in the interactive and in the editor. Error count in editor is displayed in the errors windows and we can build and then see errors there. Editors tests follow this pattern.

However, as far as I can see, we cannot follow this patter for interactive solution. We should find errors in the interactive (scripting) window, I could not find an easy way to do so. I would prefer to solve this problem with a separate code review. That is why I created a new issue and added TODOs for it. I think that the priority of this verification is low because we already verify error messages within those tests.

Please let me know if you prefer to arrange this other way.